In tests, put processing speed in an output file (i.e. xls) instead of printing it on the terminal

mvandermade opened the pull request:
Deze pull request resolved de volgende punten van issue #30:

CSV file per test voor tijden van runnen en resultaat.
Niet overschrijven van csv file.
Mappen dot, png, csv aanmaken als ze er nog niet zijn.

Heb de main.py aangepast zodat er logging op plaatsvindt.
Als deze PR is gemerged kunnen de andere tests worden aangepast om naar CSV te schrijven.
(issue #30 openstaande punten)

Nog een sidenote:

Ik had de functie os.chdir() nodig om de paden goed te krijgen.
 input_output/file_output.py r.18
```
os.chdir(os.path.dirname(os.path.abspath(__file__)))
ROOT = os.path.abspath('../')
os.chdir(ROOT)
```


Dit zorgt ervoor dat het PythonPath de map is waar main.py staat.

mvandermade committed:
Update main.py

test loop to demonstrate the csv writer

The test is responsible for time.time() to time the test.
First a file is appended, the returned filename is then used to append to the file for each test result.
Each row holds a separate test result

mvandermade committed:
Update file_output.py

Initial commit functions create_csv_file and write_csv_line

mvandermade committed:
Update .gitignore

output_files instead of output_graph for output

mvandermade committed:
Update file_output.py

fixed the amounts of newline written to the csv each writerow()
Changed formatting of file to YYYYMMDD-HH_MM_SS-name

mvandermade committed:
Update main.py

added csv logging to main

mvandermade committed:
Merge branch 'master' into csvwriter-for-tests

mvandermade committed:
Update main.py

Additional comments and automatically create output dir for dot files

mvandermade committed:
Merge branch 'master' into csvwriter-for-tests

JKleinRot commented on input_output/file_output.py:
```diff
@@ -50,4 +54,40 @@ def save_graph_in_png(filename):
+    example: YYYYMMDD-HH_MM_SS-name
+    The path of the csv file, including extension .csv, is returned as String.
+    If the folder output_files/csv does not exist it is created.
+    The argument newline = '' to prevens double carriage returns in non-binary mode.
```
prevents

JKleinRot commented on input_output/file_output.py:
```diff
@@ -50,4 +54,40 @@ def save_graph_in_png(filename):
+    """
+    Appends a csv line to a file.
+    Uses Excel compatible delimiter, quotechar and quoting.
+    The argument newline = '' to prevens double carriage returns in non-binary mode.
```
prevents

JKleinRot commented on input_output/file_output.py:
```diff
@@ -50,4 +54,40 @@ def save_graph_in_png(filename):
+    """
+
+    csv_filename = ROOT + '/output_files/csv/' + time.strftime('%Y%m%d-%H_%M_%S') + '-' + name + '.csv'
+    os.makedirs(os.path.dirname(csv_filename), exist_ok=True)
```
Ik weet niet waarom deze regel erin staat, maar zonder werkt het ook.

mvandermade replied:
De regel zorgt ervoor dat als de map waar de csv naar toe wordt geschreven wordt aangemaakt als deze niet bestaat.

Heb het aangepast in de comments.

JKleinRot commented on main.py:
```diff
@@ -2,6 +2,9 @@
 from supporting_components.graph_io import *
```
Misschien dat we iets met deze filenaam moeten. Het is eigenlijk geen main meer. Misschien kun je van deze file dan meer iets maken van dat je gewoon de csvwriter test met de color refinement zoals dat nu gebeurt. En het dan ook in de testmap zetten.

VeraPrinsen replied:
Ja hier ben ik het heel erg mee eens. Eigenlijk wat onze main eerst was, zijn nu de tests in de tests map en main gebruiken we eigenlijk niet meer.

mvandermade replied:
ik heb hem verplaatst naar de tests map

VeraPrinsen commented on main.py:
```diff
@@ -12,9 +15,34 @@
 graphs = load_graph_list(filename)
 
-os.makedirs(os.path.dirname('output_files/main/'), exist_ok=True)
+# Test output csv creation
```
Zeker om deze reden denk ik dat deze file dan maar de csv writer test moet worden, aangezien dat hier letterlijk wordt getest

VeraPrinsen commented on main.py:
```diff
@@ -12,9 +15,34 @@
+# main test loop
 for i in range(0, len(graphs)):
     outputfilename = 'output_files/main/' + file + '_' + str(i) + '.dot'
+    os.makedirs(os.path.dirname(outputfilename), exist_ok=True)
```
Is er een manier om dit 1 keer buiten de loop te doen? Dit lijkt me een beetje redundant

mvandermade replied:
weggehaald, wordt anders dubbel gedaan in save_graph_as_dot

VeraPrinsen commented on main.py:
```diff
@@ -12,9 +15,34 @@
     outputfilename = 'output_files/main/' + file + '_' + str(i) + '.dot'
+    os.makedirs(os.path.dirname(outputfilename), exist_ok=True)
+
+    start_time: float = time.time()
```
Is het nodig om het type te noemen? Ik dacht dat time() van zichzelf al gewoon een float was. Gewoon op onnodige code uit het programma te houden

VeraPrinsen commented on main.py:
```diff
@@ -12,9 +15,34 @@
+    os.makedirs(os.path.dirname(outputfilename), exist_ok=True)
+
+    start_time: float = time.time()
     with open(outputfilename, 'w') as g0:
```
Deze kan nog gerefactored worden naar save_graph_as_dot van onze eigen file_output

VeraPrinsen commented on main.py:
```diff
@@ -12,9 +15,34 @@
     with open(outputfilename, 'w') as g0:
         write_dot(color_refinement(degree_color_initialization(graphs[i])), g0)
+
+    test_result = True
```
Ik zie in deze test nergens waar test_result False kan zijn. Is het dan nodig om dit toe te voegen?

mvandermade replied:
ik heb een uitgebreidere test toegevoegd, waar ik ook de csv weer lees om de schrijfactie te verifieren.

mvandermade committed:
Update main.py

remove typing

mvandermade committed:
Update main.py

refactor save_graph_as_dot
unused import

mvandermade committed:
Update main.py

remove makedirs, it is already taken care of by save_graph_as_dot

mvandermade committed:
Update csvwriter.py

has moved to tests

mvandermade committed:
Update file_output.py

typos

mvandermade committed:
Update file_output.py

mvandermade committed:
Update csvwriter.py

made it clearer that the test is a mock. purpose is to demonstrate that a csv file is outputted.

mvandermade committed:
Update csvwriter.py

test the write action by reading back data from disk.
ignoring the first row of the csv file in the test.
